Clustered-cache-configuration-state.asciidoc

```diff
@@ -0,0 +1,73 @@
+Clustered cache configuration / state
+=====================================
+
+Currently all members of a cluster manage their configuration, cache lifecycle and most of the state independently.
+This means that in order for a cache to be created across a cluster, each node must define its configuration locally and
+start it. There is also no validation that a cache configuration is compatible across nodes. Also, while some
+configuration attributes can be changed at runtime, this does not happen cluster-wide (although it might make sense for
+some configuration attributes to be asymmetric, e.g. capacity factor). Additionally, when new nodes join a running
+cluster, they should also start any clustered caches so that symmetry is maintained.
+
+Implementation
+--------------
+
+* Add a ___state cache handled by the InternalCacheRegistry. This cache will contain both cache configurations (stored
+as strings using the Infinispan XML schema) and a list of running caches (do we need to store more state ?).
+* Introduce a new ClusteredCacheContainer API (see below)
+* Cache configuration and lifecycle performed via the old DefaultCacheManager API will continue to behave as it does
+currently (i.e. each node is independent)
+* Allow configuration Attributes to be either Global or Local.
+* Implement a variant of equals (equalsIgnoreLocal ?) for Configuration objects to validate congruency ignoring local
+attributes.
+* Support the +start+ configuration attribute for caches (which can be either +LAZY+ or +EAGER+) so that EAGER caches
```

Comment:
EAGER/LAZY comes from WildFly. In server it has always been ignored, since we never had a way to start a cache across the cluster when accessed through the remote endpoint. In embedded mode, it was never a thing